### PR TITLE
Workaround for Windows AMI conda install step failing

### PR DESCRIPTION
**Pull request by atalman**
We get a failure during this step:
https://github.com/pytorch/test-infra/blob/main/aws/ami/windows/scripts/Installers/Install-Miniconda3.ps1#L31
``
Miniconda installation failed, no hook found
``
Workaround it to install Miniconda first to get past this error and continue installing everything else.

### CODE REVIEW

File: aws/ami/windows/windows.pkr.hcl [hunk 1]
```diff
@@ -40,6 +40,20 @@ source "amazon-ebs" "windows_ebs_builder" {
 build {
   sources = ["source.amazon-ebs.windows_ebs_builder"]
 
+  # Install conda, it needs to be installed under SYSTEM to avoid this broken
+  # installation https://github.com/ContinuumIO/anaconda-issues/issues/11799.
+  # Also this needs to come after all the tools are installed to avoid error
+  # CondaHTTPError: HTTP 000 CONNECTION FAILED when connecting to conda (?)
```
Comment (ZainRizvi) [inline aws/ami/windows/windows.pkr.hcl:46]: Is this part of the comment no longer accurate, where it says it needs to be installed after all the other tools?

The issue implies that the installation actually fails but still appears to have succeeded. How would we test this?

**Review by ZainRizvi — COMMENTED** [review]
How do we test that the AMI works with this change, especially given the bug mentioned in the comments

File: aws/ami/windows/windows.pkr.hcl (hunk 1 above)
Comment (atalman) [reply aws/ami/windows/windows.pkr.hcl:46]: Right looks like its not accurate anymore. let me fix it

File: aws/ami/windows/windows.pkr.hcl [hunk 2]
```diff
@@ -40,6 +40,20 @@ source "amazon-ebs" "windows_ebs_builder" {
 build {
   sources = ["source.amazon-ebs.windows_ebs_builder"]
 
+  # Install conda, it needs to be installed under SYSTEM to avoid this broken
+  # installation https://github.com/ContinuumIO/anaconda-issues/issues/11799.
+  # Also this needs to come after all the tools are installed to avoid error
```
Comment (atalman) [inline aws/ami/windows/windows.pkr.hcl:45]:
```suggestion
```

File: aws/ami/windows/windows.pkr.hcl [hunk 3]
```diff
@@ -40,6 +40,19 @@ source "amazon-ebs" "windows_ebs_builder" {
 build {
   sources = ["source.amazon-ebs.windows_ebs_builder"]
 
+  # Install conda, it needs to be installed under SYSTEM to avoid this broken
+  # installation https://github.com/ContinuumIO/anaconda-issues/issues/11799.
+  # CondaHTTPError: HTTP 000 CONNECTION FAILED when connecting to conda (?)
```
Comment (atalman) [inline aws/ami/windows/windows.pkr.hcl:45]:
```suggestion
```

**Review by ZainRizvi — APPROVED** [review]

**Comment by ZainRizvi** [comment]
Let's keep an eye out for the `HTTP 000 CONNECTION FAILED when connecting to conda` failure in case it reappears during testing

**Pull request merged**